soil_api/models/soil_type.py

```diff
@@ -0,0 +1,109 @@
+from __future__ import annotations
+
+from enum import Enum
+from typing import List
+
+from pydantic import BaseModel, Field
+
+from soil_api.models.shared import BoundingBoxGeometry, FeatureType, PointGeometry
+
+
+class SoilTypes(Enum):
```

Comment:
When retrieving the value of a pixel from the raster, the returned value is an integer between [0, 29] to represent each soil type. t0, t1, etc. isn't domain specific, but allows to map each integer value to the correct soil type name. Do you suggest to instead store this mapping `{0: "Acrisols", 1: "Albeluvisols", ... }` as a dict? If so, where? In `soil_type.py`, or the `config` / `settings` file?

Reply:
You could use an IntEnum or an Enum, with the integers as values

```python
class SoilType(Enum):
    acrisols = 0
    albeluvisols = 1
    ...
```

then you could get the codes using

```python
SoilType(0).name  # Returns "acrisols"
```
You could also use an `dict`, although I would consider just using a `list` if the integers are always going to be ∈ [0, 29]. If you don't need validation using Pydantic, I think using a `dict` or a `list` is a better solution.

I'd rather have them defined in `soil_type.py` than in the settings-object, I don't think it makes as much sense to use the settings-object for things that should not be set using environment variables

Reply:
I like the approach of using the integers as values, but then I would no longer be able to use `SoilTypes` as the type for the `most_probable_soil_type` attribute, because the API should return the name of the soil type (not the integer code). Here is how it is defined now:

```
class SoilTypeInfo(BaseModel):
    most_probable_soil_type: SoilTypes = Field(
        ...,
        description="The most probable soil type at the queried location",
        example="Acrisols",
    )
```
Here is the error when trying to feed `SoilType(integer_val).name` to `SoilTypeInfo`:
```
pydantic_core._pydantic_core.ValidationError: 1 validation error for SoilTypeInfo
most_probable_soil_type
  Input should be 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29 or 255 [type=enum, input_value='Podzols', input_type=str]
```


Should `most_probable_soil_type` be set to `str` instead to solve this? Like this:

```
class SoilTypeInfo(BaseModel):
    most_probable_soil_type: str = Field(
        ...,
        description="The most probable soil type at the queried location",
        example="Acrisols",
    )
```
It would sort of defeat the purpose of defining the enum for use in the schema though?

Reply:
I think I understand it a bit better now. You need a way to get the correct string based on integers, but you also want to have a type that can validate if the input strings-value are correct. I see that you use this method to convert integers to the correct soil-type:

```python
soil_type=SoilTypes.__members__.get(f"t{key}", SoilTypes.No_information),
```

I think I would still prefer the `SoilType(str, Enum)`-approach to validate incoming strings, and then convert it to an integer some other way. One idea I had is to create a dict mapping soil-types to integers. This dict could be made programmatically using the `SoilType`-enum like so:

```python
soil_code_dict = {x: i for (i, x) in enumerate(SoilTypes)}
```

Downside with this is that we depend on the enumeration order of the Enum, which as far as I know always should be the order of which the enum-values appear in the Enum. Still not ideal

Reply:
If you only need to serialize to string on the way out, you could perhaps use a field_serializer instead:

https://stackoverflow.com/questions/67911340/encode-pydantic-field-using-the-enum-name-instead-of-the-value

Reply:
That solves it!